pattern-library/sass/patterns/_tables.scss

```diff
@@ -23,3 +24,148 @@ $table-border-color:                        palette(grayscale, x-light) !default
 // ----------------------------
```

Comment:
NIT: We usually break up the Sass partials a bit more based on different types/extensions of patterns (see the [Grid partial](https://github.com/edx/ux-pattern-library/blob/master/pattern-library%2Fsass%2Fpatterns%2F_grid.scss) as an example). It makes it easier to abstract and scan. Mind going through your different cases and syncing that up?